### ISSUE TITLE
`linked` config setting not working as expected?

### ISSUE THREAD

**Issue by ianstormtaylor** [changesets/changesets#570]
## Affected Packages

`pkg:cli`

## Problem

I was assuming `linked` would help with enforcing packages to always use the same version amongst each other, similar to how Lerna does. However that appears to not be the case. Here's the recent versioning pull request on Slate that shows the issue: https://github.com/ianstormtaylor/slate/pull/4198

Both `slate` and `slate-react` are bumped to `0.62.1` but the others are left at `0.62.0`.

## Proposed solution

Either change `linked` to keep things at the same version, or potentially add another setting called `pinned` or something that does it. The use case being keeping a set of packages always on the same version when one of them is bumped so that people can be guaranteed to have packages that work together really easily. (I believe this is what popular projects like Babel and React do?)

**Comment by rijkvanzanten** [comment]
@Andarist any insight around this? I was hoping we could rely on `"linked": [["*"]]` to link every package within the monorepo together similar to how Lerna would work using [`lerna version --force-publish`](https://github.com/lerna/lerna/tree/main/commands/version#--force-publish)

**Comment by glasser** [comment]
Agreed that this would be useful — `changesets` looks awesome and I'd love to migrate Apollo Server from lerna to it, but we depend on being able to publish all of our "entry point" packages (apollo-server-express, apollo-server-lambda, etc) at the same version so we can talk about v3.1.2 of Apollo Server in a simple way.

**Comment by dbismut** [comment]
@ianstormtaylor Not sure I'm understanding this correctly, but I stumbled upon this issue because I believe I had the same problem. I'm maintaining [@use-gesture](https://github.com/pmndrs/use-gesture) with the following package structure 👍 

```
@use-gesture/core
@use-gesture/react
@use-gesture/vanilla
```

I wanted both `@use-gesture/react` and `@use-gesture/vanilla` to be strictly dependent on `@use-gesture/core` versioning, but it seemed that patching `@use-gesture/core` wouldn't update neither of them.

I realized that in my `package.json` files for both `react` and `vanilla` package, I had:

```
"dependencies": {
  "@use-gesture/core": "^10.0.0"
}
```

... instead of:

```
"dependencies": {
  "@use-gesture/core": "10.0.0"
}
```

Moving to a strict versioning made it work as expected (ie patching `core` resulted in all packages to bump versions).

Hope it helps and sorry if this is unrelated.
